las_dispersion_scan/dscan.py

```diff
@@ -682,14 +682,30 @@ def run(
 
         while remaining and not self._stop:
             idx, setpoint = remaining[0]
-            self.stage.set(setpoint).wait(timeout=timeout)
+            #self.stage.set(setpoint).wait(timeout=timeout)
+            self.stage.set(setpoint)
             t0 = time.monotonic()
             while not self._stop and time.monotonic() - t0 < dwell_time:
                 time.sleep(dwell_time / 10.0)
 
             if self._stop:
                 break
 
+            n = 0
+            while abs(self.stage.wm()-setpoint)>0.01: # Then try again
```

Comment:
An arbitrary `0.01` that doesn't take into account the stage's units isn't ideal...

I feel that this loop should be part of the positioner move, or a separate routine to do this motion in a single call with a timeout, or ...

Stating it another way, I should be able to use `set()` on a buggy/faulty positioner and expect it to get to its positions with any retries required within the provided timeout. Baking the logic into the scan isn't where it should go, in my opinion.

Reply:
Yeah, I agree. That was my kludgy fix that ended up working pretty well. 

My thinking was that the scan assumes everything is in mm. Any half-decent stage should be able to reach its setpoint within 10um or better. When dealing with a less-than-half-decent stage (approximately 25% decent in the case of the elliptec) then being off by 10um is a sign that the motion didn't complete all the way. 

When using timeouts, the scan just fails every time when using an Elliptec motor. Generally issuing another move command fixed the problem in one try.

Reply:
I'm working on a reusable version of this that can be shoved in the positioner class itself; should be ready by the afternoon